packages/alea-frontend/components/QuizDashboard.tsx

```diff
@@ -298,6 +334,22 @@ const QuizDashboard: NextPage<QuizDashboardProps> = ({ courseId }) => {
               manuallySetPhase,
               problems,
             } as QuizWithStatus;
+
+            if (!isNew && stats.totalStudents > 0) {
+              const originalProblems = selectedQuiz?.problems || {};
+              const validation = validateQuizUpdate(
+                originalProblems,
+                problems,
+                stats.totalStudents
+              );
+
+              if (!validation.valid) {
+                alert(`Cannot update quiz: ${validation.reason}`);
+                setIsUpdating(false);
+                return;
+              }
+            }
```

Comment:
can this `if` check be skipped? 
```suggestion
          const isValid = !isNew && validateQuizUpdate(
            selectedQuiz?.problems || {},
            problems,
            stats.totalStudents
          );

          if (!validation.valid) {
            alert(`Cannot update quiz: ${validation.reason}`);
            setIsUpdating(false);
            return;
          }
```